src/InvoiceGenerator/InvoiceGenerator.BusinessLogic/ClientService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace InvoiceGenerator.BusinessLogic
+{
+    public class ClientService
+    {
+        public List<ClientViewModel> GetClients()
+        {
+            using (var context = new InvoiceDBContext()) // using statement is discarded once the squiggly brackets have been passed, saves memory
+            {
+                context.Database.EnsureCreated();   // Ensures database is created
+
+                var clients = context.Clients.ToList(); // Assign a bunch of client objects to a list
+
+                var viewModels = new List<ClientViewModel>();
+
+                foreach (var db in clients)
+                {
+                    viewModels.Add(ClientViewModel.FromDbModel(db)); // Add the client details to a list
+                }
+                return viewModels;
+            }
+        }
+
+        public void AddClients(string clientName, string clientAddress, string contactName, string contactEmail)
```

Comment:
Might be worth taking in the ViewModel here, and using it when constructing the new Client instance on line 34.